### ISSUE TITLE
Sort RefBy fields logically

### ISSUE THREAD

**Issue by samm82** [JacquesCarette/Drasil#3567]
As noted in #3259 (see [this comment](https://github.com/JacquesCarette/Drasil/issues/3259#issuecomment-1650336592)), the output FR shows up first in the list in the RefBy field (the following screenshot is from [IM:calOfLandingTime from the Drasil website](https://jacquescarette.github.io/Drasil/examples/projectile/SRS/srs/Projectile_SRS.html#IM:calOfLandingTime)):
![image](https://github.com/JacquesCarette/Drasil/assets/35857611/ab7753d9-00b8-4b23-9b93-ace6824ab65b)

I would think that the IMs would be first, then the FRs in the order they appear; the order of these should follow some logic. Maybe the ordering of the columns/rows in the traceability matrices could be used as a reference for how to sort them?

**Comment by balacij** [comment]
Yep, it's just sorted by `UID` right now if I remember correctly. I think @JacquesCarette suggest sorting by visible label in #2468.

Now that I see it, the labelling/abbreviation convention doesn't seem uniform. IM labels follow camelCase while the (N)FRs use Upper-Kebab-Case.

**Comment by JacquesCarette** [comment]
I definitely would not sort a `Sentence` but rather make sure our `Ref` data-structure is sortable by something that will correspond to visible label.

**Comment by JacquesCarette** [comment]
We should take a look at what features of `Sentence` are actually used in the short names that appear in `Reference`. I am guessing very few. That would let us do the same kind of refactor I did with `NounPhrase`. Probably quite a bit easier, in fact.
